google/resource_container_cluster.go

```diff
@@ -1079,43 +1086,52 @@ func resourceContainerClusterCreate(d *schema.ResourceData, meta interface{}) er
-		LegacyAbac: &containerBeta.LegacyAbac{
+		NetworkPolicy:                  expandNetworkPolicy(d.Get("network_policy")),
+	}
+
+	if d.Get("enable_autopilot").(bool) {
+		cluster.Autopilot = &containerBeta.Autopilot{
+			Enabled:         d.Get("enable_autopilot").(bool),
+			ForceSendFields: []string{"Enabled"},
+		}
+	} else {
+		cluster.InitialNodeCount = int64(d.Get("initial_node_count").(int))
+		cluster.MaintenancePolicy = expandMaintenancePolicy(d, meta)
+		cluster.InitialClusterVersion = d.Get("min_master_version").(string)
+		cluster.LegacyAbac = &containerBeta.LegacyAbac{
 			Enabled:         d.Get("enable_legacy_abac").(bool),
 			ForceSendFields: []string{"Enabled"},
-		},
-		LoggingService:          d.Get("logging_service").(string),
-		MonitoringService:       d.Get("monitoring_service").(string),
-		NetworkPolicy:           expandNetworkPolicy(d.Get("network_policy")),
-		AddonsConfig:            expandClusterAddonsConfig(d.Get("addons_config")),
-		EnableKubernetesAlpha:   d.Get("enable_kubernetes_alpha").(bool),
-		IpAllocationPolicy:      ipAllocationBlock,
-		PodSecurityPolicyConfig: expandPodSecurityPolicyConfig(d.Get("pod_security_policy_config")),
-		Autoscaling:             expandClusterAutoscaling(d.Get("cluster_autoscaling"), d),
-		BinaryAuthorization: &containerBeta.BinaryAuthorization{
+		}
+		cluster.MonitoringService = d.Get("monitoring_service").(string)
+		cluster.AddonsConfig = expandClusterAddonsConfig(d.Get("addons_config"))
+		cluster.EnableKubernetesAlpha = d.Get("enable_kubernetes_alpha").(bool)
+		cluster.PodSecurityPolicyConfig = expandPodSecurityPolicyConfig(d.Get("pod_security_policy_config"))
+		cluster.Autoscaling = expandClusterAutoscaling(d.Get("cluster_autoscaling"), d)
+		cluster.BinaryAuthorization = &containerBeta.BinaryAuthorization{
 			Enabled:         d.Get("enable_binary_authorization").(bool),
 			ForceSendFields: []string{"Enabled"},
-		},
-		ShieldedNodes: &containerBeta.ShieldedNodes{
+		}
+		cluster.ShieldedNodes = &containerBeta.ShieldedNodes{
 			Enabled:         d.Get("enable_shielded_nodes").(bool),
 			ForceSendFields: []string{"Enabled"},
-		},
-		ReleaseChannel: expandReleaseChannel(d.Get("release_channel")),
-		EnableTpu:      d.Get("enable_tpu").(bool),
-		NetworkConfig: &containerBeta.NetworkConfig{
+		}
+		cluster.NetworkConfig = &containerBeta.NetworkConfig{
 			EnableIntraNodeVisibility: d.Get("enable_intranode_visibility").(bool),
 			DefaultSnatStatus:         expandDefaultSnatStatus(d.Get("default_snat_status")),
 			DatapathProvider:          d.Get("datapath_provider").(string),
-		},
-		MasterAuth:     expandMasterAuth(d.Get("master_auth")),
-		ResourceLabels: expandStringMap(d, "resource_labels"),
+		}
+		cluster.ReleaseChannel = expandReleaseChannel(d.Get("release_channel"))
+		cluster.EnableTpu = d.Get("enable_tpu").(bool)
+		cluster.MasterAuth = expandMasterAuth(d.Get("master_auth"))
+		cluster.ResourceLabels = expandStringMap(d, "resource_labels")
 	}
```

Comment:
Its seems a little off to ignore half the parameters if gke auto pilot is enabled. This would result in a continuous diff if the user does define those parameters in the terraform configuration. Instead of refactoring this piece can you add look into utilizing the `ConflictsWith` keyword on the resource schema ?

Reply:
adding @c2thorn who is more familiar with auto_pilot

Reply:
+1 to `ConflictsWith` on incompatible fields. Example here: https://github.com/hashicorp/terraform-provider-google/blob/master/google/resource_container_cluster.go#L326